src/centralstore/incremental/PropertyLink.cpp

```diff
@@ -0,0 +1,186 @@
+        return newAddress;
+    }
+}
+
+/**
+ * Create a brand new property link to an empty node block
+ *
+ * */
+PropertyLink* PropertyLink::create(std::string name, char value[]) {
+    unsigned int nextAddress = 0;
+    char dataName[PropertyLink::MAX_NAME_SIZE] = {0};
+    strcpy(dataName, name.c_str());
+    unsigned int newAddress = PropertyLink::nextPropertyIndex * PropertyLink::PROPERTY_BLOCK_SIZE;
+    PropertyLink::propertiesDB->seekp(newAddress);
+    PropertyLink::propertiesDB->write(dataName, PropertyLink::MAX_NAME_SIZE);
+    PropertyLink::propertiesDB->write(reinterpret_cast<char*>(value), PropertyLink::MAX_VALUE_SIZE);
+    if (!PropertyLink::propertiesDB->write(reinterpret_cast<char*>(&nextAddress), sizeof(nextAddress))) {
+        property_link_logger.error("Error while inserting the property = " + name +
+                                   " into block a new address = " + std::to_string(newAddress));
+        return NULL;
+    }
+    PropertyLink::propertiesDB->flush();
+    PropertyLink::nextPropertyIndex++;  // Increment the shared property index value
+    return new PropertyLink(newAddress, name, value, nextAddress);
+}
+
+/**
+ * If a nextPropAddress available return PropertyLink with that else return null pointer
+ * */
+PropertyLink* PropertyLink::next() {
+    if (this->nextPropAddress) {
+        return new PropertyLink(this->nextPropAddress);
+    }
+    return NULL;
+}
+
+bool PropertyLink::isEmpty() { return !(this->blockAddress); }
+
+PropertyLink* PropertyLink::get(unsigned int propertyBlockAddress) {
+    PropertyLink* pl = NULL;
+    if (propertyBlockAddress > 0) {
+        char propertyName[PropertyLink::MAX_NAME_SIZE] = {0};
+        char propertyValue[PropertyLink::MAX_VALUE_SIZE] = {0};
+        unsigned int nextAddress;
+        PropertyLink::propertiesDB->seekg(propertyBlockAddress);
+
+        if (!PropertyLink::propertiesDB->read(reinterpret_cast<char*>(&propertyName), PropertyLink::MAX_NAME_SIZE)) {
+            property_link_logger.error("Error while reading property name from block = " +
+                                       std::to_string(propertyBlockAddress));
+        }
+        if (!PropertyLink::propertiesDB->read(reinterpret_cast<char*>(&propertyValue), PropertyLink::MAX_VALUE_SIZE)) {
+            property_link_logger.error("Error while reading property value from block = " +
+                                       std::to_string(propertyBlockAddress));
+        }
+
+        if (!PropertyLink::propertiesDB->read(reinterpret_cast<char*>(&(nextAddress)), sizeof(unsigned int))) {
+            property_link_logger.error("Error while reading property next address from block = " +
+                                       std::to_string(propertyBlockAddress));
+        }
+        pl = new PropertyLink(propertyBlockAddress, std::string(propertyName), propertyValue, nextAddress);
+    }
+    return pl;
+}
+const unsigned long PropertyLink::PROPERTY_BLOCK_SIZE =
```

Comment:
This type of constant definition can be done in a header file.

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/ed31ba0af33e9629173808960eef3cfe90c5dbe9

Reply:
Done